src/server.cpp

```diff
@@ -2874,8 +2874,14 @@ std::wstring Server::handleChat(const std::string &name, const std::wstring &wna
 				L"It was refused. Send a shorter message";
 	}
 
+	auto message = wide_to_utf8(wmessage);
+	auto first = message.find_first_of("\n\r");
+	if (first != std::wstring::npos && !trim(message.substr(first)).empty()) {
```

Comment:
Neither `\n` nor `\r` are sent in regular chat messages. Thus latter check is not just superfluous, but would also allow players to add empty lines in the chat (after their message).

Reply:
The latter check is for trailing new lines. It could be trimmed instead

Reply:
The trim function does not modify `message`. Please correct me when I missed something but a message like `hello world\n` would pass this check and be sent to the other players, including the additional new line.